### ISSUE TITLE
Crash at SkyrimSE.exe+014CAAD: DoDelayed runs GetNodeByName on a detached thread and races the engine's lazy bone creation (double free of a NiNode child array)

### ISSUE THREAD

**Issue by TiGRoNdev** [ModiLogist/TheNewGentleman#18]
**Version:** 4.2.4 (the shipped dll + pdb); `Common::BaseUtil::DoDelayed` is unchanged on master (4.2.6).
**Game:** SSE 1.6.1170, SKSE 2.2.6, Engine Fixes with its default memory patch.

Related: #14 (suspected race in `Core::UpdateActor`) and #15 (experimental main-thread delayed update). This is a *proven* instance of that race with the exact engine mechanism, the exact stack, and a reproducer — and it is not `AddKeyword` but the scene graph.

## What happens

`Core::SetActorSize` (Core.cpp:312) calls `ut->DoDelayed(lambda, condition, 0, true, ...)` for every actor on `Load3D` and on every addon/armor update. When the condition is not immediately true, `DoDelayed` (BaseUtil.cpp:130) spawns `std::thread([=]{ ... }).detach()` which polls `condition()` every 500 ms — and `condition()` is

```cpp
[actor]() -> bool { return actor && actor->Is3DLoaded() && actor->GetNodeByName(Common::genBoneNames[Common::egbBase]); }
```

`actor->GetNodeByName(...)` on a `BSFlattenedBoneTree` skeleton is **not a read**. `BSFlattenedBoneTree::GetObjectByName` (1.6.1170: `SkyrimSE.exe+0D30380`) resolves the name to a bone index and calls `GetBoneNode(index, create = true)` (`+0D300D0`), which, when the bone's `NiNode` has not been materialised yet, **builds it**: recursively creates the parent bone's node, allocates a `NiNode`, `parent->AttachChild(node, true)`, and stores it in the bone table slot. There is no lock in that path.

Two threads asking for the same bone at the same time — two `DoDelayed` threads for the same actor (one per `Load3D`, one per equip), or one of them and the game thread — both find the slot empty and both build. Two `AttachChild` calls grow the same parent's child array concurrently: both free the old buffer, the allocator hands the same block out twice, and the second owner's first element ends up where the first owner's `new[]` element count lives. The next `~NiNode` walks the children by that count — a pointer — and dies at `SkyrimSE.exe+014CAAD` (`mov rcx,[rcx]`), usually minutes later, in whoever happens to release the tree last (FSMP's `ActorManager`, the engine's animation graph manager, the end-of-update release queue). Exactly the "corrupt memory, crash much later somewhere random" shape described in #14.

## Evidence

* 19 crash logs (15–16 Sep 2026), all at `+014CAAD`; in every one of them `R8` (the element count read from `data-8`) is a valid `NiNode*` — a skeleton bone with a parent and `parentIndex == -1` — in 7 of 19 an unnamed node under `GenitalsScrotumLag`.
* A race detector (MinHook on `NiNode::AttachChild`/`InsertChildAt`/`DetachChild*`/`SetAt`/`SetSize`/both destructors, recording the thread inside each node's child array) caught **three threads at once inside `AttachChild` on `GenitalsScrotumLag`** of one NPC, all three with this stack (symbolised with the shipped PDB):

  ```
  SkyrimSE.exe+0D30190  BSFlattenedBoneTree::GetBoneNode  (AttachChild of the freshly built bone)
  SkyrimSE.exe+0D303AD  BSFlattenedBoneTree::GetObjectByName
  SkyrimSE.exe+0D1D9EC  NiNode::GetObjectByName
  TheNewGentleman.dll+2A2A2  `Core::SetActorSize'::<lambda_1>::operator()   Core.cpp:312
  TheNewGentleman.dll+56D9F  `Common::BaseUtil::DoDelayed'::<lambda_1>::operator()   BaseUtil.cpp:153
  TheNewGentleman.dll+5940E  std::thread::_Invoke<...>
  TheNewGentleman.dll+9FB4A  thread_start
  ```

  Eight overlaps in one 10-minute run, all eight identical.
* Serialising `GetBoneNode`'s create path with a recursive mutex (from another SKSE plugin — fast path unchanged, lock only when the slot is empty, re-check after taking it) removes the crash completely: **5/5 clean runs of a stress loop that crashed 12/12 before**. That lock logs every "a second thread asked for the bone while it was being built" event: 30–200 per 10-minute run, every single one from this thread, always `NPC GenitalsScrotum [GenScrot]` (`genBoneNames[egbScrot]`).

Reproducer: any load order with TNG + a mod that reloads many actors at once (we disable/enable a dozen NPCs every 3 s from a plugin; the game's own cell loads do the same thing more slowly, which is why it looks like "random crashes every other evening").

## Suggested fix

Don't touch the scene graph from a thread. Run both the condition and the lambda on the game thread — e.g. `SKSE::GetTaskInterface()->AddTask(...)` that re-queues itself until the condition holds (or the retry budget is spent), instead of `std::thread(...).detach()` — which is what #15 did. The same applies to the immediate path (`fixedDelay == 0 && condition()` → `func()` synchronously): it runs on whatever thread called `SetActorSize`, and the `Load3D` hook can be on a loader thread.
